web/modules/custom/contact_information/src/Plugin/Validation/Constraint/EmailConstraint.php

```diff
@@ -2,13 +2,18 @@
 
 declare(strict_types = 1);
 
-namespace Drupal\joinup_core\Plugin\Validation\Constraint;
+namespace Drupal\contact_information\Plugin\Validation\Constraint;
```

Comment:
This is not the `RequiredIfOtherFieldHasValue` validation constraint, this is `EmailConstraint`. This has been placed specifically into the `contact_information` because it is not considered to be reusable except for 1 single case inside the contact informtaion module.

This only exists because of historical reasons, there was a specific demand for this regex by a functional contractor who did not know that Drupal ships with a very robust email validation and researched this solution. If we put this in `joinup_validation` then we risk that people will use it instead of the email validation that is built into Drupal core.

To make it more clear that this should be avoided I documented it in the validator as well as in the code that calls it, and included references to the original ticket that introduced it:

```
/**
 * Checks if an email is valid according Joinup specifications.
 *
 * Do not use this for anything other than contact information emails. For all
 * other emails use the standard EmailValidator.
 *
 * @see \Drupal\Component\Utility\EmailValidator
 */
```

```
function contact_information_entity_bundle_field_info_alter(&$fields, EntityTypeInterface $entity_type, $bundle) {
  // During the development of the contact information functionality we were
  // helpfully supplied by the business analysts with a regular expression to
  // validate the email addresses, and they were insistent to use this. So we
  // use a bespoke regex instead of \Drupal\Component\Utility\EmailValidator.
  // @see https://webgate.ec.europa.eu/CITnet/jira/browse/ISAICP-2404
  // ...
}
```

I think we could even remove this since the contractor left the project years ago, and replace it with the core `EmailConstraint`.